### ISSUE TITLE
Dfget is downloading large file (5GB), and out of memory or killed (trigger system OOM) appears

### ISSUE THREAD

**Issue by kivis-online** [dragonflyoss/Dragonfly#1206]
## Question

OS：ubuntu 16.0.4 / CentOS 7.3   CPU:2C Mem:4G
dfget version:1.0.0

1. The first client node downloads a new 5G file: no error is reported, the time is longer (more than 8 minutes), it is estimated that the speed is slower,
2. Immediately after the second and third client nodes download the same 5GB file: from the peer (P2P network takes effect) in the log, about 30s ~ 1min, dfget error out of memory or killed (system OOM) . After investigation, cpu.iowait was 45%, the system memory was insufficient.
    # /home/work/Dragonfly/dfget -u http: //x.x.x.x/test --node x.x.x.x: 8002 -b --locallimit 10G --totallimit 10G
3. try to lower the limit to 200M, dfget no longer reports errors, memory usage is stable at more than 1G, but the download speed is really slow.
    # /home/work/Dragonfly/dfget -u http: //x.x.x.x/test --node x.x.x.x: 8002 -b --locallimit 200M --totallimit 200M
4. Use wget http: //x.x.x.x/test to download quickly, it takes more than 1mins. No crashes occurred.

In summary, ask if dfget has similar memory usage control parameters, which can prevent the program from crashing directly due to memory consumption（base on current "dfget --help"，I did not find any memory usage control parameters）, because it is really fast with ordinary wget download, but no crash occurs.

**Comment by lowzj** [comment]
Could you paste some crashed logs?
